src/hooks/useDeposits.ts

```diff
@@ -167,24 +172,38 @@ export function useUserDeposits(
         deposits.map((d) => d.depositTxHash)
       );
 
-      // If the Scraper API indexed the optimistically added deposit,
-      // then we need to remove it from local storage.
-      const indexedLocalDeposits = localUserDeposits.filter(
-        (localPendingDeposit) =>
-          indexedDepositTxHashes.has(localPendingDeposit.depositTxHash)
-      );
-      removeLocalDeposits(
-        indexedLocalDeposits.map((deposit) => deposit.depositTxHash)
-      );
-
       // If the Scraper API is still a few blocks behind and didn't index
       // the optimistically added deposits, then we merge them to provide instant
       // visibility of a deposit after a user performed a transaction.
-      const notIndexedLocalDeposits = localUserDeposits.filter(
-        (localPendingDeposit) =>
-          !indexedDepositTxHashes.has(localPendingDeposit.depositTxHash)
+      // - If the local deposit is pending, we show it if it's not indexed yet.
+      // - If the local deposit is filled, we only show it if the indexed deposit is pending.
+      const localDepositsToShow = localUserDeposits.filter(
+        (localUserDeposit) => {
+          if (localUserDeposit.status === "pending") {
+            return !indexedDepositTxHashes.has(localUserDeposit.depositTxHash);
+          } else if (localUserDeposit.status === "filled") {
+            const indexedDeposit = deposits.find(
+              (d) => d.depositTxHash === localUserDeposit.depositTxHash
+            );
+            return indexedDeposit ? indexedDeposit.status === "pending" : true;
```

Comment:
To remove the branch

```suggestion
            return !indexedDeposit || indexedDeposit.status === "pending";
```